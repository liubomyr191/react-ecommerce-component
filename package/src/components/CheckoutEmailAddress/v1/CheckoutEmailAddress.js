import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { addTypographyStyles, applyTheme } from "../../../utils";

const StyledDiv = styled.div`
  ${addTypographyStyles("CheckoutEmailAddress", "labelText")}
  border-bottom-color: ${applyTheme("checkoutEmailAddressBorderBottomColor")};
  border-bottom-style: solid;
  border-bottom-width: ${applyTheme("checkoutEmailAddressBorderBottomWidth")};
  border-left-color: ${applyTheme("checkoutEmailAddressBorderLeftColor")};
  border-left-style: solid;
  border-left-width: ${applyTheme("checkoutEmailAddressBorderLeftWidth")};
  border-right-color: ${applyTheme("checkoutEmailAddressBorderRightColor")};
  border-right-style: solid;
  border-right-width: ${applyTheme("checkoutEmailAddressBorderRightWidth")};
  border-top-color: ${applyTheme("checkoutEmailAddressBorderTopColor")};
  border-top-style: solid;
  border-top-width: ${applyTheme("checkoutEmailAddressBorderTopWidth")};
  color: ${applyTheme("checkoutEmailAddressColor")};
  padding-bottom: ${applyTheme("checkoutEmailAddressPaddingBottom")};
  padding-left: ${applyTheme("checkoutEmailAddressPaddingLeft")};
  padding-right: ${applyTheme("checkoutEmailAddressPaddingRight")};
  padding-top: ${applyTheme("checkoutEmailAddressPaddingTop")};
`;

const StyledSpan = styled.span`
  ${addTypographyStyles("CheckoutEmailAddressEmphasis", "labelTextBold")}
`;

class CheckoutEmailAddress extends Component {
  static propTypes = {
    emailAddress: PropTypes.string.isRequired,
    isAccountEmail: PropTypes.bool.isRequired
  };

  renderAccountEmail = () => {
    const { isAccountEmail } = this.props;

    if (isAccountEmail) {
      return "Signed in as";
    }

    return null;
  };

  render() {
    const { emailAddress } = this.props;
    return (
      <StyledDiv>
        {this.renderAccountEmail()} <StyledSpan>{emailAddress}</StyledSpan>
      </StyledDiv>
    );
  }
}

export default CheckoutEmailAddress;
