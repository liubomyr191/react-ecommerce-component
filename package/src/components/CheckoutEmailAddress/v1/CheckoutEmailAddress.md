### CheckoutEmailAddress

### Overview
Displays a customers email address inside the checkout flow

#### Basic usage when user has an account
```jsx
<CheckoutEmailAddress
  emailAddress="account.email@example.com"
  isAccountEmail={true}
/>
```

#### Basic usage as a guest
```jsx
<CheckoutEmailAddress
  emailAddress="guest.email@example.com"
  isAccountEmail={false}
/>
```
