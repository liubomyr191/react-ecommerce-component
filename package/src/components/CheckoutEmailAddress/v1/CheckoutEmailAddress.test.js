import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import CheckoutEmailAddress from "./CheckoutEmailAddress";

test("render email address of a user with an account", () => {
  const component = shallow(<CheckoutEmailAddress emailAddress="account.email@example.com" isAccountEmail={true} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("render email address of a guest", () => {
  const component = shallow(<CheckoutEmailAddress emailAddress="guest.email@example.com" isAccountEmail={false} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
