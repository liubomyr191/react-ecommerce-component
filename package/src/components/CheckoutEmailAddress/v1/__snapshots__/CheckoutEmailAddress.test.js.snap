// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render email address of a guest 1`] = `
.c0 {
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
}

.c1 {
  font-weight: 700;
}

<div
  className="c0"
>
   
  <span
    className="c1"
  >
    guest.email@example.com
  </span>
</div>
`;

exports[`render email address of a user with an account 1`] = `
.c0 {
  color: #3c3c3c;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
}

.c1 {
  font-weight: 700;
}

<div
  className="c0"
>
  Signed in as
   
  <span
    className="c1"
  >
    account.email@example.com
  </span>
</div>
`;
